### PR TITLE
parser: show error for `abstract`, `intrinsic` and `native` constructors

### LINKED ISSUE #2452: specifying an abstract constructor feature should result in an error
e.g.

```
ex is
  something is abstract
```

### PR DESCRIPTION
**Pull request by simonvonhackewitz**
fix #2452

### CODE REVIEW

File: src/dev/flang/ast/AstErrors.java [hunk 1]
```diff
@@ -2209,6 +2209,14 @@ public static void constructorWithReturnType(SourcePosition pos)
         " or remove the return type if you want to define a constructor.");
   }
 
+  public static void unimplementetConstructor(SourcePosition pos, String keyword)
```
Comment (michaellilltokiwa) [inline src/dev/flang/ast/AstErrors.java:2212]: unimplementetConstructor => unimplementedConstructor

File: src/dev/flang/ast/AstErrors.java [hunk 2]
```diff
@@ -2209,6 +2209,14 @@ public static void constructorWithReturnType(SourcePosition pos)
...
+  {
+    error(pos, keyword + " feature must not be constructor.",
+        "A constructor feature must always be implemented. " +
```
Comment (michaellilltokiwa) [inline src/dev/flang/ast/AstErrors.java:2215]:
```suggestion
        "A constructor feature must always be implemented.\n " +
```

File: src/dev/flang/parser/Parser.java [hunk 3]
```diff
@@ -3389,11 +3389,8 @@ List<Cond> invariant(boolean atMinIndent)
   /**
    * Parse implRout
    *
-implRout    : "is" "abstract"
-            | ARROW "abstract"
-            | "is" "intrinsic"
+implRout    : ARROW "abstract"
```
Comment (michaellilltokiwa) [inline src/dev/flang/parser/Parser.java:3392]: I think we should leave the grammar as is. Parser does parse e.g. `is abstract` even though it then emits an error.

**Comment by fridis** [comment]
please change the title, the parser should never `throw` any errors but produce / display / show errors.

File: src/dev/flang/parser/Parser.java (hunk 3 above)
Comment (fridis) [reply src/dev/flang/parser/Parser.java:3392]: I think we shoud do both: change the grammar (which becomes simpler) and change the parser to not accept `is abstract`, etc. Then, we might not need new error messages and everything becomes easier...

**Pull request merged**